src/circuit/gadget/sinsemilla.rs

```diff
@@ -210,3 +210,164 @@ where
+            config: Self::Config,
+        ) -> Result<(), Error> {
+            let mut layouter = SingleChipLayouter::new(cs)?;
+            let ecc_chip = EccChip::construct(config.0);
+
+            // The two `SinsemillaChip`s share the same lookup table.
+            SinsemillaChip::<K, MAX_WORDS>::load(config.1.clone(), &mut layouter)?;
+
+            // This MerkleCRH example is purely for illustrative purposes.
+            // It is not an implementation of the Orchard protocol spec.
+            {
+                let chip1 = SinsemillaChip::<K, MAX_WORDS>::construct(config.1);
+
+                let merkle_crh = HashDomain::new(
+                    chip1.clone(),
+                    ecc_chip.clone(),
+                    &SinsemillaHashDomains::MerkleCrh,
+                );
+
+                // Left leaf
+                let left = {
+                    let left: Vec<Option<bool>> =
+                        (0..250).map(|_| Some(rand::random::<bool>())).collect();
+                    let left = Message::from_bitstring(
+                        chip1.clone(),
+                        layouter.namespace(|| "witness left"),
+                        left,
+                    )?;
+                    let left = merkle_crh.hash_to_point(layouter.namespace(|| "left"), left)?;
+                    let left = left.extract_p();
+                    chip1.witness_message_piece_field(
+                        layouter.namespace(|| "witness left piece"),
+                        left.inner().value(),
+                        25,
+                    )?
+                };
+
+                // Right leaf
+                let right = {
+                    let right: Vec<Option<bool>> =
+                        (0..250).map(|_| Some(rand::random::<bool>())).collect();
+                    let right = Message::from_bitstring(
+                        chip1.clone(),
+                        layouter.namespace(|| "witness right"),
+                        right,
+                    )?;
+                    let right = merkle_crh.hash_to_point(layouter.namespace(|| "right"), right)?;
+                    let right = right.extract_p();
+                    chip1.witness_message_piece_field(
+                        layouter.namespace(|| "witness left piece"),
+                        right.inner().value(),
+                        25,
+                    )?
+                };
+
+                // Layer 0
+                let l = {
+                    let merkle_depth_orchard = 32;
+                    let layer = 0;
+                    let l = pallas::Base::from_u64(merkle_depth_orchard - 1 - layer);
+                    chip1.witness_message_piece_field(layouter.namespace(|| "l"), Some(l), 1)?
+                };
+
+                // Parent
+                let parent = Message::from_pieces(chip1, vec![l, left, right]);
+                merkle_crh.hash_to_point(layouter.namespace(|| "parent"), parent)?;
```

Comment:
If possible, witness the expected result and constrain it to equal the output of this.